packages/react-meteor-data/suspense/useTracker.ts

```diff
@@ -0,0 +1,303 @@
+    )
+  }
+}
+
+export const cacheMap = new Map<string, Entry>()
+
+interface Entry {
+  deps: EJSON[]
+  promise: Promise<unknown>
+  result?: unknown
+  error?: unknown
+}
+
+// Used to create a forceUpdate from useReducer. Forces update by
+// incrementing a number whenever the dispatch method is invoked.
+const fur = (x: number): number => x + 1
+const useForceUpdate = () => useReducer(fur, 0)[1]
+
+export type IReactiveFn<T> = (c?: Tracker.Computation) => Promise<T>
+
+export type ISkipUpdate<T> = <T>(prev: T, next: T) => boolean
+
+interface TrackerRefs {
+  computation?: Tracker.Computation
+  isMounted: boolean
+  trackerData: any
+}
+
+function resolveAsync<T>(key: string, promise: Promise<T> | null, deps: DependencyList = []): typeof promise extends null ? null : T {
+  const cached = cacheMap.get(key)
+  useEffect(() =>
+    () => {
+      setTimeout(() => {
+        if (cached !== undefined && isEqual(cached.deps, deps)) cacheMap.delete(key)
+      }, 0)
+    }, [cached, key, ...deps])
+
+  if (promise === null) return null
+
+  if (cached !== undefined) {
+    if ('error' in cached) throw cached.error
+    if ('result' in cached) {
+      const result = cached.result as T
+      setTimeout(() => {
+        cacheMap.delete(key)
+      }, 0)
+      return result
+    }
+    throw cached.promise
+  }
+
+  const entry: Entry = {
+    deps,
+    promise: new Promise((resolve, reject) => {
+      promise
+        .then((result: any) => {
+          entry.result = result
+          resolve(result)
+        })
+        .catch((error: any) => {
+          entry.error = error
+          reject(error)
+        })
+    })
+  }
+  cacheMap.set(key, entry)
+  throw entry.promise
+}
+
+export function useTrackerNoDeps<T = any>(key: string, reactiveFn: IReactiveFn<T>, skipUpdate: ISkipUpdate<T> = null): T {
+  const { current: refs } = useRef<TrackerRefs>({
+    isMounted: false,
+    trackerData: null
+  })
+  const forceUpdate = useForceUpdate()
+
+  // Without deps, always dispose and recreate the computation with every render.
+  if (refs.computation != null) {
+    refs.computation.stop()
+    // @ts-expect-error This makes TS think ref.computation is "never" set
+    delete refs.computation
+  }
+
+  // Use Tracker.nonreactive in case we are inside a Tracker Computation.
+  // This can happen if someone calls `ReactDOM.render` inside a Computation.
+  // In that case, we want to opt out of the normal behavior of nested
+  // Computations, where if the outer one is invalidated or stopped,
+  // it stops the inner one.
+  Tracker.nonreactive(() =>
+    Tracker.autorun(async(c: Tracker.Computation) => {
+      refs.computation = c
+
+      const data: Promise<any> = Tracker.withComputation(c, async() => await reactiveFn(c))
+      if (c.firstRun) {
+        // Always run the reactiveFn on firstRun
+        refs.trackerData = data
+      } else if (!skipUpdate || !skipUpdate(await refs.trackerData, await data)) {
```

Comment:
I'm not sure about these `await`s, but it's hard to come up with a direct problem too. 🤷